packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -811,7 +821,7 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
   @override
   void initState() {
     super.initState();
-    _textEditingController = widget.textEditingController ?? TextEditingController();
+    _textEditingController = widget.textEditingController ?? TextEditingController(text: widget.initialValue);
```

Comment:
Maybe set the selection to the end of the string, or expose `TextEditingValue` instead? IIRC an invalid selection can spell trouble if the text input is auto-focused (also I'm planning to stop painting the caret in `RenderEditable` if an invalid selection is specified: https://github.com/flutter/flutter/pull/79607).

Reply:
I opted to expose TextEditingValue